datafusion/datasource-arrow/src/source.rs

```diff
@@ -66,6 +71,8 @@ enum ArrowFormat {
 pub(crate) struct ArrowStreamFileOpener {
     object_store: Arc<dyn ObjectStore>,
     projection: Option<Vec<usize>>,
+    projected_schema: Option<SchemaRef>,
+    schema_adapter_factory: Option<Arc<dyn SchemaAdapterFactory>>,
```

Comment:
this seems like a nice improvement (the Arrow readers now get a SchemaAdapterFactory, so they should now also handle schema evolution)

Do we need to add tests for this anywhere? Perhaps a slt showing we can read multiple arrow files with different schemas?